### ISSUE TITLE
Don't terminate interpreter on read error

### ISSUE THREAD

**Issue by mcandre** [YosysHQ/yosys#3020]
## Steps to reproduce the issue

Please forgive my bad SystemVerilog syntax, I'm just experimenting.

```console
$ cat half_adder.v
module half_adder;
    function ha(input a, b, output y, carry);
        y = a ^ b
        carry = a & b
    endfunction

    initial begin
        y, carry = ha(0, 0)

        $display("ha(0, 0): %0d, %0d", y, carry);
    end

    // abc
endmodule

$ yosys

yosys> read -sv half_adder.v
1. Executing Verilog-2005 frontend: half_adder.v
Parsing SystemVerilog input from `half_adder.v' to AST representation.
half_adder.v:4: ERROR: syntax error, unexpected TOK_ID

$
```

## Expected behavior

In most REPL's, when you try to load program code into the interpreter, and the code fails to parse, then the interpreter displays an error message but then resumes prompting for further commands. For example, this is a common workflow when actively debugging your program.

## Actual behavior

yosys terminates the REPL. This requires the user to manually relaunch the yosys interpreter each time a parse error is introduced in their code.

**Comment by thcoura** [comment]
I totally agree with this enhancement or "request". Quitting every time when an error occurs for me is a huge limitation to the tool
